docs/topics/wasm/wasm-debugging.md

```diff
@@ -0,0 +1,149 @@
+   ```
+
+   >The port number can vary because the 8080 port may be unavailable. You can find the actual port number printed
+   > in the Gradle build console.
+   >
+   {type="tip"}
+
+   You can see a "Click me!" button. Click it:
+
+   ![Click me](wasm-composeapp-browser-clickme.png){width=650}
+
+   Now you can see the Compose Multiplatform logo:
+
+   ![Compose app in browser](wasm-composeapp-browser.png){width=650}
+
+## Debug in the browser
+
+> Currently, debugging is only available in the browser. Debugging in 
+> [IntelliJ IDEA](https://youtrack.jetbrains.com/issue/KT-64683/Kotlin-Wasm-debugging-in-IntelliJ-IDEA) and 
+> [Fleet](https://youtrack.jetbrains.com/issue/KT-64684) is in the plan. 
+>
+{type="note"}
+
+You can debug this Compose Multiplatform application
+in your browser out of the box without additional configurations. 
+
+However, for other projects, you may need to set the required configuration in the Gradle 
+build file. For more information, see [Configuration for debugging](#configuration-for-debugging).
+
+> For this tutorial, we are using Chrome. If you use a different browser, the experience should be similar and the following 
+> steps should also work. 
+> 
+{type="tip"}
+
+To debug a Kotlin/Wasm application:
+
+1. In the browser window of the application, access developer tools by right-clicking and selecting the **Inspect** action.
+Alternatively, you can use the **F12** shortcut or select **View** | **Developer** | **Developer Tools**.
+
+2. Switch to the **Sources** tab and select the Kotlin file to debug. In this tutorial, we'll work with the `Greeting.kt` file.
+
+3. Set breakpoints in the Kotlin file by clicking on the numbers of the code lines you want to inspect. Only the code lines 
+with darker numbers accept breakpoints.
+
+![Set breakpoints](wasm-breakpoints.png){width=700}
+
+4. Interact with the application by clicking on the **Click me!** button. This action triggers the execution of the 
+code, and the debugger pauses when the execution reaches a breakpoint.
+
+5. Inspect variables and code execution at the breakpoints by using debugging control buttons such as: 
+   * Step into to investigate a function deeper
+   * Step over to execute the current line and pause on the next line
+   * Step out to execute the code until it exits the current function
+
+![Debug controls](wasm-debug-controls.png){width=700}
+
+6. Check the **Call stack** and **Scope** tool windows to trace the sequence of function calls and pinpoint the location of any errors.
+
+![Check call stack](wasm-debug-scope.png){width=700}
+
+7. Make the required changes to the code and [run the application](#run-the-application) again.
+
+### Configuration for debugging
+
+By default, browsers lack access to the project sources necessary for debugging. To gain access, you can configure the Webpack DevServer 
+to serve these sources. Add the following code snippet to the Gradle build file inside the `ComposeApp` directory:
+
+```kotlin
```

Comment:
Can you separate this into two code samples? Then the user can easily copy each snippet into the right place in the file.